pesto-cli/pesto/common/testing/service_manager.py

```diff
@@ -86,17 +88,30 @@ def run(self):
             self.pull()
             logger.info("Starting container with {} on port {}".format(self.docker_image, self.host_port))
 
-            self._container = self.CLIENT.containers.run(
-                self.docker_image,
-                ports={self.service_port: self.host_port},
-                detach=True,
-                remove=True,
-                runtime="nvidia" if self.nvidia else None,
-                volumes={self.image_volume_path: {
-                    "bind": self.host_volume_path,
-                    "mode": "rw",
-                }},
-            )
+            if self.network is None :
+                self._container = self.CLIENT.containers.run(
+                    self.docker_image,
+                    ports={self.service_port: self.host_port},
+                    detach=True,
+                    remove=True,
+                    runtime="nvidia" if self.nvidia else None,
+                    volumes={self.image_volume_path: {
+                        "bind": self.host_volume_path,
+                        "mode": "rw",
+                    }},
+                )
+            else:
+                self._container = self.CLIENT.containers.run(
+                    self.docker_image,
+                    network=self.network,
```

Comment:
Why the `ports` mapping has disappeared ? 
Side note: according to [this documentation](https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.ContainerCollection.run), the "host" network mode is incompatible with `ports` option.

Reply:
I thought that all network mode were incompatible with ports option, not only "host" network. I will change that condition.

Reply:
Network behavior: when network is None, network is set to "bridge" (Create a new network stack for the container on on the bridge network). This change fix the issue encounters when running docker container with network and ports set: https://github.com/CS-SI/pesto/pull/2/commits/1b750e9ebb0f88d9034057748679debe45a3ef2b.